cfme/tests/containers/test_container_provider_crud.py

```diff
@@ -0,0 +1,71 @@
+import fauxfactory
+import pytest
+
+from cfme.common.provider_views import ContainerProvidersView
+from cfme.containers.provider import ContainersProvider
+from cfme.fixtures.provider import setup_or_skip
+from cfme.utils.appliance.implementations.ui import navigate_to
+
+
+pytestmark = [
+    pytest.mark.tier(1),
+    pytest.mark.provider([ContainersProvider], scope='function')
+]
+
+
+@pytest.mark.tier(3)
+def test_container_provider_crud(request, appliance, provider):
+
+    """
+    Polarion:
+        assignee: juwatts
+        caseimportance: medium
+        casecomponent: Containers
+        initialEstimate: 1/6h
+    """
+
+    if provider.exists:
+        provider.delete()
+
+    setup_or_skip(request, provider)
+
+    request.addfinalizer(lambda: provider.delete_if_exists(cancel=False))
+
+    view = appliance.browser.create_view(ContainerProvidersView)
+
+    success_add = '{} Providers "{}" was saved'.format(provider.string_name, provider.name)
+
+    view.flash.assert_success_message(success_add)
+
+    assert provider.exists
+
+    provider_edit_view = navigate_to(provider, 'Edit')
```

Comment:
Providers are `Updateable`, is there a reason you're explicitly filling the edit view here?

Its forcing you to use `new_provider_name` in the flash assert following, and then you're setting the name attribute outside of the `Updateable` context manager (L56).

Reply:
updated to use the `Updateable` context manager.